File: Java/1863_Sum_of_All_Subset_XOR_Totals.java
class Solution {
    int result = 0;
    boolean[] choosed;
    int[] nums;
    public int subsetXORSum(int[] nums) {
        choosed = new boolean[nums.length];
        this.nums = nums;
        
        backtracking(0, 0);
        
        return result;
    }
    private void backtracking(int index, int total){
        result += total;
        
        for(int i = index; i < nums.length; i++){
            if(choosed[i] == false){
                choosed[i] = true;
                total ^= nums[i];

                //System.out.println(total);
                backtracking(i, total);

                choosed[i] = false;
                total ^= nums[i];
            }
        }
    }
}

File: CS/1863_Sum_of_All_Subset_XOR_Totals.cs
public class Solution {
    int result = 0;
    int[] nums;
    public int SubsetXORSum(int[] nums) {
        this.nums = nums;
        for(int i = 0; i < nums.Length; i++){
            for(int c = 1; i+c <= nums.Length; c++){
                helper(c, 1, 0^nums[i], i);
            }
        }
        return result;
    }
    private void helper(int count, int curCount, int res, int index){
        if(count == curCount){
            result += res;
            return;
        }
        for(int i = index+1; i < nums.Length; i++){
            helper(count, curCount+1, res^nums[i], i);
        }
    }
}
